src/auto/vfs.rs

```diff
@@ -82,10 +83,50 @@ impl<O: IsA<Vfs>> VfsExt for O {
         }
     }
 
-    //#[cfg(any(feature = "v2_50", feature = "dox"))]
-    //fn register_uri_scheme<'a, 'b, 'c, 'd, P: Into<Option<&'a /*Unimplemented*/VfsFileLookupFunc>>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>, R: Into<Option<&'b /*Ignored*/glib::DestroyNotify>>, S: Into<Option<&'c /*Unimplemented*/VfsFileLookupFunc>>, T: Into<Option</*Unimplemented*/Fundamental: Pointer>>, U: Into<Option<&'d /*Ignored*/glib::DestroyNotify>>>(&self, scheme: &str, uri_func: P, uri_data: Q, uri_destroy: R, parse_name_func: S, parse_name_data: T, parse_name_destroy: U) -> bool {
-    //    unsafe { TODO: call ffi::g_vfs_register_uri_scheme() }
-    //}
+    #[cfg(any(feature = "v2_50", feature = "dox"))]
+    fn register_uri_scheme<P: Fn(&Vfs, GString) -> File + 'static, Q: Into<Option<P>>, R: Fn(&Vfs, GString) -> File + 'static, S: Into<Option<R>>>(&self, scheme: &str, uri_func: Q, parse_name_func: S) -> bool {
+        let uri_func = uri_func.into();
+        let parse_name_func = parse_name_func.into();
+        let uri_func_data: Box_<Option<P>> = Box::new(uri_func.into());
+        unsafe extern "C" fn uri_func_func<P: Fn(&Vfs, GString) -> File + 'static, R: Fn(&Vfs, GString) -> File + 'static>(vfs: *mut ffi::GVfs, identifier: *const libc::c_char, user_data: glib_ffi::gpointer) -> *mut ffi::GFile {
```

Comment:
And passed by reference then of course

Reply:
Oh it is `from_glib_borrow()` but then passed by value. That's wrong and unsafe, it will cause crashes :) `from_glib_borrow()` return values must only be passed by reference to other code and never ever leave the scope where they were created.

Reply:
For signal closures we use `&str` for the parameters and `String` for return values btw. See e.g. `connect_format_entry_text()` from `gtk::ComboBox`.